featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,242 @@
+        prim.name
+        for prim in get_valid_primitives(
+            entityset,
+            target_dataframe_name,
+            1,
+            selected_trans_primitives,
+        )[1]
+    ]
+
+    recommended_primitives.update(
+        _recommend_non_numeric_primitives(
+            entityset,
+            target_dataframe_name,
+            valid_primitive_names,
+        ),
+    )
+    skew_numeric_primitives = set(["square_root", "natural_logarithm"])
+    valid_skew_primtives = skew_numeric_primitives.intersection(valid_primitive_names)
+
+    valid_time_series_primitives = set(TIME_SERIES_PRIMITIVES).intersection(
+        valid_primitive_names,
+    )
+
+    numeric_tags_only_df = entityset[target_dataframe_name].ww.select("numeric")
+
+    if valid_skew_primtives:
+        recommended_primitives.update(
+            _recommend_skew_numeric_primitives(
+                numeric_tags_only_df,
+                valid_skew_primtives,
+            ),
+        )
+
+    if valid_time_series_primitives:
+        recommended_primitives.update(valid_time_series_primitives)
+    return list(recommended_primitives)
+
+
+def _recommend_non_numeric_primitives(
+    entityset: EntitySet,
+    target_dataframe_name: str,
+    valid_primitives: List[str],
+) -> set:
+    """Get a set of non-numeric primitives for a given dataset and a list of primitives.
+
+    Description:
+        Given a single table entity set with a `target_dataframe_name` and an applicable list of `valid_primitives`,
+        get a set of primitives which produce non-unique features.
+
+    Args:
+        entityset (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `entityset`.
+        valid_primitives (List[str]): List of primitives to calculate and check output features.
+    """
+
+    recommended_non_numeric_primitives = set()
+    # Only want to run feature generation on non numeric primitives
+    numeric_columns_to_ignore = list(
+        entityset[target_dataframe_name]
+        .ww.select(include="numeric", return_schema=True)
+        .columns,
+    )
+    features = dfs(
+        entityset=entityset,
+        target_dataframe_name=target_dataframe_name,
+        trans_primitives=valid_primitives,
+        max_depth=1,
+        features_only=True,
+        ignore_columns={target_dataframe_name: numeric_columns_to_ignore},
+    )
+
+    for f in features:
+        if (
+            f.primitive.name is not None
+            and f.primitive.name not in recommended_non_numeric_primitives
+        ):
+            try:
+                matrix = calculate_feature_matrix([f], entityset)
+                for f_name in f.get_feature_names():
+                    if len(matrix[f_name].unique()) > 1:
+                        recommended_non_numeric_primitives.add(f.primitive.name)
+            except Exception:  # If error in calculating feature matrix pass on the recommendation
```

Comment:
minor styling question, but in general, do we prefer this approach or `contextlib.suppress`?

Reply:
I've always just used `try,except`.

Reply:
Do we know of a specific case where this Exception gets thrown or are we just including this in case this happens? Just thinking that if this step fails, that probably means we have another bug somewhere that we need to address.